typescript.js

```diff
@@ -9,15 +9,19 @@ module.exports = {
       parserOptions: { warnOnUnsupportedTypeScriptVersion: false },
       extends: ['plugin:@typescript-eslint/recommended'],
       rules: {
-        'no-use-before-define': 'off', // Turning off because we have the TS version off
-        '@typescript-eslint/no-use-before-define': 'off',
-        '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
-
+        'no-use-before-define': 'off', // @typescript-eslint/no-use-before-defined requires this to be off
+        '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
```

Comment:
Note from Jordi:
@typescript-eslint/no-use-before-define -- from off to error but does not check functions because functions are hoisted so it is safe (https://eslint.org/docs/latest/rules/no-use-before-define#options) We have the nomral version turned on in es6's rules, so just turning on the typescript one that builds on that.